### PR TITLE
Add integrate-docs github action

### PR DESCRIPTION
**Pull request by jamonation**
### What should this PR do?
This PR adds a reusable GitHub action to enforce documentation from Google Cloud Storage buckets and render it into the Academy site.

It will fetch the following:
- `changelog.md`
- `chainctl` autogenerated docs
- Enforce events
- Enforce OpenAPI `api.json` specification

### Why are we making this change?
At the moment updates to these pages are done on an ad-hoc basis. This action should eventually allow for automatic PRs and merges of docs in the future.

### What are the acceptance criteria? 
Docs on the netlify preview site won't appear up to date, since the action runs separately from that site. This action should get tested in a separate branch from `main` and not automatically deploy anything until we know for sure that it works. 

### How should this PR be tested?
I'm still working on steps to preview a working version of this. I'm thinking that a parameterized workflow on a separate branch would be a good way, then deploy to a staging Cloud Run service. Will update with instructions once I have something figured out.

### CODE REVIEW

**Comment by jamonation** [comment]
Just coming back to this and I think it just needs a little extra work to:
1. only run on `workflow_dispatch`
2. switch to opening a PR against the `main` branch, that way it will render all the Enforce docs in the netlify preview

**Comment by jamonation** [comment]
I've got the github action working to automatically create a PR for changes to Enforce docs.

Screenshot 2023-01-19 at 2 26 37 PM

If there are no changes it results in a noop, otherwise it will open a PR. Any subsequent reruns with changes will get force-pushed, since the upstream docs from Cloud Storage ought to be considered the canonical version of things.

Here's a screenshot of an automated PR as well:

Screenshot 2023-01-19 at 2 27 28 PM

I'm sure there will be  a few issues to resolve once this is merged, but a review would be great @ltagliaferri and @erikaheidi!

**Review by ltagliaferri — APPROVED** [review]
What do you think about adding ` \` at the end of the multiline lines to break them up here?

File: .github/workflows/autodocs-enforce.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        GIT_STATUS=$(git status content --porcelain)
+        if [ -n "${GIT_STATUS}" ]; then
+          git add content/
+          git commit -am "Automated docs commit"
+          git push origin HEAD:autodocs --force
+
+          # Now open a PR if needed
+          PR=$(gh pr list --json title,headRefName,url |jq '.[] | select(.headRefName=="autodocs")')
+          if [ -z "${PR}" ]; then
+            gh pr create --assignee jamonation --base main --head autodocs --title "Enforce docs autocommit" --body "Enforce docs autocommit" --no-maintainer-edit --label automated,documentation,enforce
```
Comment (jamonation) [inline .github/workflows/autodocs-enforce.yaml:69]:
```suggestion
            gh pr create --assignee jamonation --base main --head autodocs --title "Enforce docs autocommit" \
              --body "Enforce docs autocommit" --no-maintainer-edit --label automated,documentation,enforce
```

File: .github/workflows/integrate-enforce-docs/action.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  project_id:
+    type: string
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: download enforce changelog.md from cloud storage
+      shell: bash
+      run: gcloud --quiet storage cp --project="${{ inputs.project_id }}" "${{ inputs.storage_bucket }}/enforce-changelog/changelog.md" content/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/
```
Comment (jamonation) [inline .github/workflows/integrate-enforce-docs/action.yaml:25]:
```suggestion
      run: |
        gcloud --quiet storage cp --project="${{ inputs.project_id }}" \
          "${{ inputs.storage_bucket }}/enforce-changelog/changelog.md" content/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/
```

File: .github/workflows/integrate-enforce-docs/action.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    - name: download enforce openapi api.json spec from cloud storage
+      shell: bash
+      run: gcloud --quiet storage cp --project="${{ inputs.project_id }}" "${{ inputs.storage_bucket }}/enforce-openapi/api.json" static/
+
+    - name: download enforce events.md from cloud storage
+      shell: bash
+      run: gcloud --quiet storage cp --project="${{ inputs.project_id }}" "${{ inputs.storage_bucket }}/enforce-events/events.md" content/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/chainguard-enforce-events/index.md
```
Comment (jamonation) [inline .github/workflows/integrate-enforce-docs/action.yaml:33]:
```suggestion
      run: |
        gcloud --quiet storage cp --project="${{ inputs.project_id }}" \
          "${{ inputs.storage_bucket }}/enforce-events/events.md" content/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/chainguard-enforce-events/index.md
```

File: .github/workflows/integrate-enforce-docs/action.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    - name: download chainctl docs from cloud storage
+      shell: bash
+      run: gcloud --quiet storage cp --recursive --project="${{ inputs.project_id }}" "${{ inputs.storage_bucket }}/chainctl-docs" /tmp/
```
Comment (jamonation) [inline .github/workflows/integrate-enforce-docs/action.yaml:37]:
```suggestion
      run: |
        gcloud --quiet storage cp --recursive --project="${{ inputs.project_id }}" \
          "${{ inputs.storage_bucket }}/chainctl-docs" /tmp/
```

File: .github/workflows/integrate-enforce-docs/action.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+    - name: copy changed chainctl docs into academy content directory
+      shell: bash
+      run: rsync -av --delete --exclude="_index.md" "/tmp/chainctl-docs/" "content/chainguard/chainguard-enforce/chainctl-docs/"
+
+    - name: fix chainctl-docs urls
+      shell: bash
+      run: sed 's#/chainguard/chainctl/#/chainguard/chainguard-enforce/chainctl-docs/#g' -i content/chainguard/chainguard-enforce/chainctl-docs/*.md
```
Comment (jamonation) [inline .github/workflows/integrate-enforce-docs/action.yaml:45]:
```suggestion
      run: |
        sed 's#/chainguard/chainctl/#/chainguard/chainguard-enforce/chainctl-docs/#g' \
          -i content/chainguard/chainguard-enforce/chainctl-docs/*.md
```

File: .github/workflows/integrate-enforce-docs/action.yaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+runs:
+  using: composite
+  steps:
+    - name: download enforce changelog.md from cloud storage
+      shell: bash
+      run: gcloud --quiet storage cp --project="${{ inputs.project_id }}" "${{ inputs.storage_bucket }}/enforce-changelog/changelog.md" content/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/
+
+    - name: download enforce openapi api.json spec from cloud storage
+      shell: bash
+      run: gcloud --quiet storage cp --project="${{ inputs.project_id }}" "${{ inputs.storage_bucket }}/enforce-openapi/api.json" static/
```
Comment (jamonation) [inline .github/workflows/integrate-enforce-docs/action.yaml:29]:
```suggestion
      run: |
      gcloud --quiet storage cp --project="${{ inputs.project_id }}" \
        "${{ inputs.storage_bucket }}/enforce-openapi/api.json" static/
```

**Pull request merged**